cxx4opencl/diff2openclc.txt

```diff
@@ -120,10 +124,47 @@ label:  // invalid: jumping forward over declaration of n
    // ...
 ----------
 
-==== Use of Clang Blocks
+==== OpenCL specific difference
+
+This section describes where {cpp} for OpenCL differs from OpenCL C
+in OpenCL specific behavior.
+
+===== Variadic macros
+
+{cpp} for OpenCL eliminates restriction on variadic macros from `OpenCL C v2.0 s6.9.e`.
+Variadic macros can be used normally as per {cpp}17 `[cpp.replace]`.
+
+===== Predefined macros
+
+Predefined version macro `__OPENCL_C_VERSION__` and macro `NULL`,
+described in `OpenCL C v2.0 s6.10`, are not supported.
+
+The following new predefined macros are added in {cpp} for OpenCL:
+
+ * `__OPENCL_CPP_VERSION__` set to value `100`.
+ * `__CL_CPP_VERSION_1_0__` also set to `100` and can be used for
+   convenience instead of a literal.
+
+===== Atomic operations
+
+{cpp} for OpenCL relaxes restriction from `OpenCL C v2.0 s6.13.11` to
+atomic types allowing them to be used by builtin operators, and not
+only by builtin functions.
+
+The behavior of operators on atomic types follows the logic from {cpp}17
+sections `[atomics.types.int]` `[atomics.types.pointer]`
+`[atomics.types.float]`.
+
+[source,c]
+----------
+atomic_int acnt;
+acnt++; // equivalent to atomic_fetch_add_explicit(&acnt, 1);
+----------
+
+===== Use of Clang Blocks
 
 Clang Blocks that are defined by the Objective-C language are not supported
 and their use can be replaced by lambdas ({cpp}17 `[expr.prim.lambda]`).
 
-This means that builtin functions using blocks, such as `enqueue_kernel`,
+The above implies that builtin functions using blocks, such as `enqueue_kernel`,
```

Comment:
To me the original wording was stronger, I assume that those builtins really don't work.

Reply:
Yes, they are not going to be accepted in C++ for OpenCL mode.